// packer-unit/grunt_config/karma.js
var grunt = require('grunt');

var karmaconfig = {
    karma: {
        options: {
            basePath: '',
            frameworks: ['jasmine'],
            files: [
                'bower_components/angular/angular.js',
                'bower_components/angular-*/angular-*.js',
                'src/app/*.js',
                'src/app/**/*.js',
                'test/unit/**/*.js'
            ],
            browsers: [
                'Chrome'
            ],
            reporters: ['progress', 'coverage'],
            preprocessors: {
              'src/app/*.js': ['coverage'],
              'src/app/*/*.js': ['coverage']
            },
            coverageReporter: {
              type : 'html',
              dir : './coverage/'
            }
        },
        unit: {
            options: {
                singleRun: true
            }
        },
        chrome: {
            options: {
                singleRun: true,
                browsers: [
                    'Chrome'
                ]
            }
        },
        phantomjs: {
            options: {
                singleRun: true,
                browsers: [
                    'PhantomJS'
                ]
            }
        },
        ie: {
            options: {
                singleRun: true,
                browsers: [
                    'IE'
                ]
            }
        },
        watch: {
            options: {
                singleRun: false,
                autoWatch: true
            }
        }
    }
}

if (grunt.file.exists('test/karma.conf.js')) {
    karmaconfig.karma.options = {
        configFile: 'test/karma.conf.js'
    };
}

module.exports = karmaconfig;
